### PR TITLE
Support arm64 architecture on macOS

### LINKED ISSUE #531: Work around libgmdjni64.dylib not compatible with arm64 architecture on macOS
Installing Gams on the arm64 version reproducing the following error.
The issue is related to the closed issue #473 
```
java.lang.UnsatisfiedLinkError: java.lang.UnsatisfiedLinkError: 
/Users/ravitb/Workspace/TAU/ilTransport/ixmp/ixmp/backend/jdbc/libgmdjni64.dylib: 
dlopen(/Users/ravitb/Workspace/TAU/ilTransport/ixmp/ixmp/backend/jdbc/libgmdjni64.dylib, 0x0001): tried: 
'/Users/ravitb/Workspace/TAU/ilTransport/ixmp/ixmp/backend/jdbc/libgmdjni64.dylib' (mach-o file, but is an incompatible 
architecture (have 'x86_64', need 'arm64')), 
'/System/Volumes/Preboot/Cryptexes/OS/Users/ravitb/Workspace/TAU/ilTransport/ixmp/ixmp/backend/jdbc/libgmdjni64.dylib' (no
such file), '/Users/ravitb/Workspace/TAU/ilTransport/ixmp/ixmp/backend/jdbc/libgmdjni64.dylib' (mach-o file, but is an 
incompatible architecture (have 'x86_64', need 'arm64'))
```

I wrote to Gams support and they suggested to replace the ``libgmdjni64.dylib`` dll file from the installation Gams path:
 
 ```
 /Library/Frameworks/GAMS.framework/Versions/Current/Resources/apifiles/Java/api/libgmdjni64.dylib
 ```
  
 to the ixmp path as follows:
 
 ```
Path to ixmp/ixmp/backend/jdbc/libgmdjni64.dylib
 ```

### PR DESCRIPTION
**Pull request by khaeru**
This PR responds to #531.

- Improve `.backend.jdbc`:
  - Pass `-Djava.library.path=…` when starting the JVM, giving two directories containing the required binary libraries (with the extension `.so` on Linux; `.dll` on Windows; and `.dylib` on macOS):
    - The …/apifiles/Java/api/ directory within the user's GAMS installation directory. 
    - The ixmp/backend/jdbc/ directory within the package itself, which contains (very old) bundled copies of the same files.
    
    The second directory is redundant if the first one is present (i.e. if a GAMS installation exists). To confirm this, **the PR currently deletes the bundled libraries entirely**. These commits could also be dropped instead.
- Expand the CI matrix to include both x86_64 and arm64 architectures on the macOS.
  - Use an updated version of iiasa/actions/setup-gams per iiasa/actions#13. This version installs an architecture-appropriate GAMS, and handles caching automatically.
  - Exclude macos-latest (arm64) for Python 3.8 and 3.9, as there are no pre-built wheels for JPype1 for this combination.   

## How to review

- Read the diff.
- Note the CI checks all pass.
- **`message_ix` usage**: Install `ixmp` from this branch on a macOS arm64 system with an arm64 version of Python. Run the test suite or other code and confirm the code functions as expected.
- **Scenario Explorer usage**: Consider implications for non-`message_ix` deployments; particularly, for the Scenario Explorer. Is GAMS installed in these contexts? Will the absence of the .dylib, .so, and/or .dll files bundled with the ixmp Python package affect these deployments?

## PR checklist

- [x] Continuous integration checks all ✅
- [x] Add or expand tests; coverage checks both ✅
- ~Add, expand, or update documentation.~
- [x] Update release notes.

### CODE REVIEW

**Comment by khaeru** [comment]
@glatterf42 let me know whether you think there is urgency to include this in the 3.9 release.

We would need to invite and wait for reviews from:
- Ravit, Sidd, and/or Setu —to check behaviour on their machines.
- Philip or someone else —to address the point about impacts on Scenario Explorer deployments.

Given this might take a few days, and the first three are probably capable to use it from `main` in the period between the 3.9 and 3.10 releases, I would lean to "no". OTOH, if we have users in the upcoming workshop who have macOS arm64 machines, this might (not necessarily) save some support overhead.

**Comment by glatterf42** [comment]
I'd also say it's likely not urgent. Workshop users will learn about GitHub anyway and if they need to use this branch, this will be a good opportunity to practice. They could also then provide feedback about their install process.
However, I think we can already invite review from our team members and if these come in very soon, we can maybe still include this in 3.9 :)

**Comment by khaeru** [comment]
Great, thanks for confirming.

**Comment by khaeru** [comment]
@glatterf42, since the other requested colleagues seem unable to review this over the past ~6 weeks, I propose we merge and wait for feedback that may come as they/others use it from `main`. If we don't get positive confirmation before we start prep for the next release, then we can ask again.

File: RELEASE_NOTES.rst
Comment (glatterf42) [inline RELEASE_NOTES.rst:7]:
```suggestion
  - :class:`.JDBCBackend` is usable on MacOS with newer, ``arm64``-architecture processors and Python/GAMS compiled for ``arm64`` (:issue:`473`, :issue:`531`).
```
Is this a typo?

**Review by glatterf42 — APPROVED** [review]
Sounds good to me, just one possible typo :)

File: RELEASE_NOTES.rst
Comment (khaeru) [reply RELEASE_NOTES.rst]: Yes, thanks! Fixed.

**Pull request merged**
